ui/src/shared/utils/useVisDomainSettings.ts

```diff
@@ -8,22 +8,38 @@ import {extent} from 'src/shared/utils/vis'
 
 /*
   This hook helps map the domain setting stored for line graph to the
-  appropriate settings on a @influxdata/giraffe `Config` object. 
+  appropriate settings on a @influxdata/giraffe `Config` object.
 
   If the domain setting is present, it should be used. If the domain setting is
   not present, then the min/max values shown should be derived from the data
   passed to the plot.
 */
+export const getValidRange = (
+  data: NumericColumnData = [],
+  startTime: number = Infinity,
+  endTime: number = -Infinity
+) => {
+  const range = extent((data as number[]) || [])
+  if (range && range.length >= 2) {
+    const start = Math.min(startTime, range[0])
+    const end = Math.max(endTime, range[1])
+    return [start, end]
+  }
+  return range
+}
+
```

Comment:
good change! note how the difficulty and oddness of testing this clued you in to a design smell - `useVisDomainSettings` might have been doing too much, both front-endy stuff with `useMemo` and calculating the valid range. the act of testing forces one to think, which can be very powerful. noice!